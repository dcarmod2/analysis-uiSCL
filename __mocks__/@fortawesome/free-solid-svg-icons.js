const I = jest.requireActual('@fortawesome/free-solid-svg-icons')

/**
 * Return just the name and not the entire SVG object. Throw an error if the
 * icon does not exist.
 */
module.exports = new Proxy(
  {},
  {
    get(target, name) {
      if (!I[name]) throw new Error(`Icon type ${name} does not exist`)
      return {
        ...I[name],
        icon: [] // blank out the exact SVG info for Snapshots
      }
    }
  }
)
