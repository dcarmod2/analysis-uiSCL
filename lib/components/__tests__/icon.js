import {faPlus} from '@fortawesome/free-solid-svg-icons'
import enzyme from 'enzyme'
import React from 'react'

import Icon from '../icon'

test('Icon should render and match snapshot', () => {
  expect(enzyme.shallow(<Icon icon={faPlus} />)).toMatchSnapshot()
})
