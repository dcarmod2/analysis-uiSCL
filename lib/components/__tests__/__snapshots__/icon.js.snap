// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render and match snapshot 1`] = `
<FontAwesomeIcon
  border={false}
  className=""
  fixedWidth={true}
  flip={null}
  icon={
    Object {
      "icon": Array [],
      "iconName": "plus",
      "prefix": "fas",
    }
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;
